File: 3042-count-prefix-and-suffix-pairs-i/3042-count-prefix-and-suffix-pairs-i.java
class Solution {
    public int countPrefixSuffixPairs(String[] words) {
        var n = words.length;
        var count = 0;
        for(var i=0; i<n-1; i++){
            var str1 = words[i];
            for(var j=i+1; j<n; j++){
                if(isPrefixAndSuffix(str1, words[j])) count++;
            }
        }
        
        return count;
    }
    
    private boolean isPrefixAndSuffix(String str1, String str2){
        return (str1.length() <= str2.length() && str2.startsWith(str1) && str2.endsWith(str1));
    }
}

File: 3042-count-prefix-and-suffix-pairs-i/3042-count-prefix-and-suffix-pairs-i.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        n = len(words)
        count = 0
        for i in range(n - 1):
            str1 = words[i]
            for j in range(i + 1, n):
                if self.is_prefix_and_suffix(str1, words[j]):
                    count += 1
        return count

    def is_prefix_and_suffix(self, str1, str2):
        return len(str1) <= len(str2) and str2.startswith(str1) and str2.endswith(str1)